### PR TITLE
Add tier1 tests for CPU hotplug MaxSockets calculation (CNV-61263)

### PR DESCRIPTION
**Pull request by guyoron1**
## Summary
- Add unit tests for the MaxSockets calculation fix that prevents vCPU count from exceeding 512
- Tests validate the setupCPUHotplug function behavior

## Test Coverage
- MaxSockets capping when topology exceeds 512 vCPU limit
- Standard topology calculation (no capping needed)
- Edge cases for 512 vCPU boundary
- Explicit MaxSockets preservation

## References
- Jira: CNV-61263
- Fix PR: kubevirt/kubevirt#14338

## Test plan
- [ ] Run tests in kubevirt CI
- [ ] Verify MaxSockets calculation matches expected values
- [ ] Validate no regression in standard CPU hotplug scenarios

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/compute/cpu_hotplug_max_sockets_test.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+This test suite validates the MaxSockets calculation fix that prevents vCPU count
+from exceeding 512 when CPU hotplug is enabled. The fix modifies setupCPUHotplug()
+in pkg/defaults/defaults.go to cap MaxSockets appropriately.
+
+The fix formula:
+  - If totalVCPUs (maxSockets * cores * threads) > 512:
+    - adjustedSockets = 512 / (cores * threads)
+    - maxSockets = max(adjustedSockets, configuredSockets)
+*/
```
Comment (0xFelix) [inline tests/compute/cpu_hotplug_max_sockets_test.go:22]: This is not a correct file header, please use the common KubeVirt header

**Comment by guyoron1** [comment]
Closing to recreate with complete test artifacts (STP, STD, stubs, implementations)
